SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -293,26 +294,23 @@ void FLocalDeploymentManager::TryStartLocalDeployment(FString LaunchConfig, FStr
 		}
 
 		// Timeout detection.
-		if (bStartingDeployment && Output.Contains(TEXT("startup completed")))
+		if (!bLocalDeploymentRunning && Output.Contains(TEXT("startup completed")))
 		{
-			bStartingDeployment = false;
+			bLocalDeploymentRunning = true;
 		}
 	});
 
 	RuntimeProcess->Launch();
 
-	while (bStartingDeployment && RuntimeProcess->Update())
-	{
-		if (RuntimeProcess->GetDuration().GetTotalSeconds() > RuntimeTimeout)
-		{
-			UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Timed out waiting for the Runtime to start."));
-			bStartingDeployment = false;
-			break;
-		}
-	}
+	// Wait for runtime to start or timeout
+	while (!bLocalDeploymentRunning && RuntimeProcess->Update() && RuntimeProcess->GetDuration().GetTotalSeconds() <= RuntimeTimeout) {}
```

Comment:
Worth at least yielding or sleeping for a small (250 microsec?) period

Reply:
`MonitoredProcess->Update()` should already have a sleep. It's what the API is for